std/format.d

```diff
@@ -2565,7 +2566,7 @@ private enum ctfpMessage = "Cannot format floating point types at compile-time";
 /*
     Floating-point values are formatted like $(REF printf, core, stdc, stdio)
  */
-private void formatValueImpl(Writer, T, Char)(auto ref Writer w, T obj, scope const ref FormatSpec!Char f)
+private void formatValueImpl(Writer, T, Char)(auto ref Writer w, T obj, scope const ref FormatSpec!Char f) @trusted
```

Comment:
I actually don't know, what triggers the need for `@trusted` here. `printFormat` itself is `@safe` (I forgot to add the keyword, but it is; btw it's also pure), `FloatingPointControl.rounding` is `@trusted` and the rest is just a copy of code that worked without `@trusted`. When all specifiers are implemented (im close to finish the work on %e) the duplicated code is not needed any more and I hope, that the need for `@trusted` will vanish too or at least it will make it easier to hunt for the place, which triggers the need.

Reply:
You should be able to figure out where the problem is pretty quick by removing it and seeing where the compiler complains.

Reply:
All I get is:

```
std/conv.d(1262): Error: @safe function std.conv.__unittest_L1259_C7 cannot call @system function std.conv.to!string.to!(double[2]).to
std/conv.d(226):        std.conv.to!string.to!(double[2]).to is declared here
```

And a dozen of similar messages, but nothing usefull.

Reply:
If https://github.com/dlang/phobos/pull/7285#pullrequestreview-326656921 is not the cause then mark `formatValueImpl` `@safe` explicitly then the compiler should complain where in `formatValueImpl` is the problem.

Reply:
This indeed points to the `buf = buftmp` stuff. I added `.dup` now and it works. But I still cannot declare `formatValueImpl` `@safe`. It tells me, that a `@safe`-function cannot call `std.range.primitives.put`. (I remember, that I got this some weeks ago too and it was the reason why I put that `@trusted` there, because it was in code, that I even did not touch.) 

Seems to be a little bit weird, like if there is some other (undocumented or somehow implicit) state than `@safe`, `@trusted` and `@system`. Could this be a compiler bug? Or is it something I don't know - maybe if one template is `@safe`, all other versions are assumed `@safe` too or something like this?

Reply:
IDK, I'm not up to figuring that out now. Might be a good idea to ask on the forum or he slack channel.

Reply:
Well, here I think we are done. With `.dup` I can remove `@trusted`. All else is an other matter that does not belong here. When I'll find the time I'll ask about this in the forum.